bufr2geojson/__init__.py

```diff
@@ -227,25 +303,62 @@ def get_qualifiers(self) -> list:
                     continue
                 if k in ID_DESCRIPTORS:
                     continue
+                if c in ("04", "05", "06"):  # , "07"):
+                    LOGGER.warning(f"Unhandled location information {k}")
                 # now remaining qualifiers
-                name = k
                 value = self.qualifiers[c][k]["value"]
                 units = self.qualifiers[c][k]["attributes"]["units"]
                 description = self.qualifiers[c][k]["description"]
                 try:
                     description = strip2(description)
                 except AttributeError:
                     pass
-                q = {
-                    "name": name,
-                    "value": value,
-                    "units": units,
-                    "description": description
-                }
-                result.append(q)
+                except Exception as e:
+                    LOGGER.error(f"{e}")
+
+                # set the qualifier value, result depends on type
+                if units in ("CODE TABLE", "FLAG TABLE"):
+                    q = {
+                        "value": value.copy()
+                    }
+                elif units == "CCITT IA5":
+                    q = {"value": description}
+                else:
+                    q = {
+                        "value": value,
+                        "units": units,
+                        "description": description
+                    }
+
+                # now assign to type of qualifier
+                if c == "01":
+                    identification[k] = q.copy()
+                if c in ("02", "03", "07", "22"):
+                    wigos_md[k] = q.copy()
+                if c in ("08", "09"):
+                    qualifiers[k] = q.copy()
+                if c == "25":
+                    processing[k] = q.copy()
+                if c == "31":
+                    associated_field[k] = q.copy()
+                if c == "33":
+                    quality[k] = q.copy()
+                if c == "35":
+                    monitoring[k] = q.copy()
+
+        result = {
+            "identification": identification,
+            "instrumentation": wigos_md,
+            "qualifiers": qualifiers,
+            "processing": processing,
+            "monitoring": monitoring,
+            "quality": quality,
+            "associated_field": associated_field
+        }
+
         return result
 
-    def get_location(self) -> Union[dict, None]:
+    def get_location(self, BUFRclass=None) -> Union[dict, None]:
```

Comment:
Annotate the type of `BUFRclass` (I would also rename to `bufr_class`).

Also add a return type.